File: TowerOfHanoi/TOH.java
package TowerOfHanoi;

import java.util.Scanner;

class TOH 
{ 
	static void towerOfHanoi(int n, char fromRod, char toRod, char auxiRod) 
	{ 
		if (n == 1) 
		{ 
			System.out.println("Move disk 1 from rod " + fromRod + " to rod " + toRod); 
			return; 
		} 
		towerOfHanoi(n-1, fromRod, auxiRod, toRod); 
		System.out.println("Move disk " + n + " from rod " + fromRod + " to rod " + toRod); 
		towerOfHanoi(n-1, auxiRod, toRod, fromRod); 
	} 
	
	public static void main(String args[]) 
	{
		Scanner scan = new Scanner(System.in);
		int numOfDisks = scan.nextInt(); // Number of disks
      	// A is fromRod and C is toRod, B is auxillary rod
		towerOfHanoi(numOfDisks, 'A', 'C', 'B');
		scan.close();
	} 
} 

File: Divide and Conquer/TOH.scala

/*Tower Of Henoi Problem*/

object Main {
    //function for the problem
    def ToH(n: Int, p: Char, q: Char, r: Char)
    {
        if(n == 1)
        {
            println("Move disk 1 from rod "+p+"to rod "+q)
        }
        ToH(n-1, p, r, q)
        println("Move disk "+n+ " from rod"+p+" to rod "+q)
        ToH(n-1, r, q, p)
    }
    //driver code
	def main(args: Array[String])
	{
		var n: Int = 4
		ToH(n, 'A', 'C', 'B')
	}
}